pkg/controller/elasticsearch/shutdown/node.go

```diff
@@ -142,13 +151,17 @@ func logStatus(logger logr.Logger, podName string, shutdown esclient.NodeShutdow
 
 // Clear deletes shutdown requests matching the type of the NodeShutdown field typ and the given optional status.
 // Depending on the progress of the shutdown in question this means either a cancellation of the shutdown or a clean-up
-// after shutdown completion.
+// after shutdown completion. Restart shutdown records will only be deleted once the corresponding node is back in the cluster.
 func (ns *NodeShutdown) Clear(ctx context.Context, status *esclient.ShutdownStatus) error {
 	if err := ns.initOnce(ctx); err != nil {
 		return err
 	}
 	for _, s := range ns.shutdowns {
 		if s.Is(ns.typ) && (status == nil || s.Status == *status) {
+			if ns.typ == esclient.Restart && !ns.nodeInCluster(s.NodeID) {
```

Comment:
Should we compare the current time with `allocation_delay` `+` `shutdown_startedmillis` in the `GET` answer in order to garbage collect shutdown requests for nodes that may never join again the cluster? (if node state is lost before the restart for example?)

Reply:
This could be a problem if there is clocks skew between the operator (node) and the nodes the Elasticsearch Pods run on. But I agree in general that we are creating garbage. Maybe we could look at the expected nodes in the cluster instead and if the restarted node is no longer expected to be in cluster remove it? I have to do some experiments.

Reply:
I realise that going by the expected nodes is more complicated than I thought because we cannot map the nodeID back to a Pod name unless we create additional state either in the operator or in the k8s API to remember which nodeID maps to which Pod name. With the help of Elasticsearch we can only do this mapping for nodes in the cluster.

Reply:
https://github.com/elastic/cloud-on-k8s/pull/5618/commits/d098f7910d93e77653541d328a8f7096142b4c3a implements your proposal.

Reply:
I have come up with another solution and updated the PR description accordingly.